### PR TITLE
global_nest_v1 support for C192 and higher

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
## Description of changes
Switch to global_nest_v1 physics suite and remove global_nest_v0. This works for C192, C384, and C768 nested, and for the non-nested C384. In other words, everything except C96. I'll have another PR soon with C96 support on top of these changes.

This is only tested on Hera. I'll tested on GAEA momentarily and update the PR description accordingly.

Output is here:

- C192: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/gnv1C192/2023022400/00L/forecast/
- C384: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/gnv1C384/2023022400/00L/forecast/
- C768: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/gnv1C768/2023022400/00L/forecast/

Ignore the C96 directory. It doesn't work yet.

## Issues addressed (optional)
FIXME: make an issue

## Contributors (optional)
None yet.

## Tests conducted
Ran the "Steve Case" on Hera

### CODE REVIEW

File: parm/WaterInTheWestC384.conf
Comment (lisa-bengtsson) [inline parm/WaterInTheWestC384.conf:1]: I couldn't see the C3 settings anywhere, could you double check that we have:

#C3 cumulus convection

imfdeepcnv=5
imfshalcnv=5

For C96, C192: 
progsigma = .false.

For C384, C768 and C1152:
progsigma = .true.
betadcu = 8.0    
betamcu = 1.5 

For C3072:
progsigma = .true.
betadcu = 1
betamcu = 1

I'm a bit uncertain about shal_cnv = .true. when we have MYNN doing shallow convection, can you check with Joe Olson? It may be needed for radiation coupling.

Comment (SamuelTrahanNOAA) [reply parm/WaterInTheWestC384.conf:1]: These are correct:

imfdeepcnv
imfshalcnv

These are absent from the namelist, so they're definitely wrong:

progsigma
betadcu
betamcu

Comment (SamuelTrahanNOAA) [reply parm/WaterInTheWestC384.conf:1]: Also, HAFS-AR doesn't really run C-anything. Those are simply reference numbers to understand the resolution. The actual C768 grid is not rotated or stretched; it is the uniform, non-rotated grid. HAFS-AR has a rotated, slightly stretched (1.0001) global grid. The nest tile is a single tile (not C-anything) at quadruple the resolution.

HAFS-AR can run uniform C384 without a nest, and C384 stretched-rotated without a nest. The other resolutions could be set up that way, but I haven't had a reason to yet. When adding nests to the global workflow, I've been using C384 resolution for testing. Hence the need for all three possibilities at C384 resolution.

**Comment by SamuelTrahanNOAA** [comment]
I've updated the paths in the PR description so they point to the forecast directories. You should find the namelists in there: input.nml and input_nest02.nml.

**Comment by SamuelTrahanNOAA** [comment]
@lisa-bengtsson - What about C1536 resolution? That's the resolution of the C384's nest.

**Comment by lisa-bengtsson** [comment]
> @lisa-bengtsson - What about C1536 resolution? That's the resolution of the C384's nest.

I have not tested this resolution, so I will give you and educated guess that we may have to tune in the future:

For C1536 use:
progsigma = .true.
betadcu = 8.0
betamcu = 1.5

**Review by AndersJensen-NOAA — APPROVED** [review]
This is good pending Mike and Joe's approval.

**Comment by SamuelTrahanNOAA** [comment]
@lisa-bengtsson - Using these three variables crashes the model: progsigma, betadcu, betamcu. I've added them to the HAFS namelists, but they're set to the GFS_typedefs defaults. (That's the same as not setting them at all.) You can re-enable them by uncommenting the relevant six lines of WaterInTheWest.conf and its resolution variants.

**Comment by lisa-bengtsson** [comment]
Can you send me the output file so I can see the nature of the crash?

**Comment by SamuelTrahanNOAA** [comment]
I'm rerunning it on Hera now to confirm it isn't a system issue. (GAEA C5 has been problematic in other ways.) Once it finishes, I'll let you know.

**Comment by SamuelTrahanNOAA** [comment]
@lisa-bengtsson I reproduced the failures on Hera. Do you know why they're failing?

#### C192 Hera
*log*: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/psigC192/2023022400/00L/hafs_forecast.log
*run directory*: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/psigC192/2023022400/00L/forecast

#### C384 Hera
*log*: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/psigC384/2023022400/00L/hafs_forecast.log
*run directory*: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/psigC384/2023022400/00L/forecast

#### C768 Hera
*log*: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/psigC768/2023022400/00L/hafs_forecast.log
*run directory*: /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/scrub/psigC768/2023022400/00L/forecast

**Comment by lisa-bengtsson** [comment]
Yes, you should add:

 prognotsitc sigmab tracer
 "TRACER", "atmos_mod", "sigmab"
           "longname",     "sigma fraction"
           "units",        "fraction"
       "profile_type", "fixed", "surface_value=0." /

after the TKE tracer in your field_table. 

Thanks!
Lisa

**Comment by SamuelTrahanNOAA** [comment]
There is one field table for both the parent and nest. That means the progsigma setting must match between parent and nest.

**Comment by lisa-bengtsson** [comment]
Ok, in that case you can set it to .true. in both. Inside the code, I do force progsigma to false if dx is large.

**Comment by SamuelTrahanNOAA** [comment]
I'm updating the scripts now and I'll let you know which resolutions run through.

**Comment by SamuelTrahanNOAA** [comment]
> Ok, in that case you can set it to .true. in both. Inside the code, I do force progsigma to false if dx is large.

I see what you mean. Yes, I'll leave in the scripts the capability of turning prognosis on or off independently for each domain. The run scripts will add sigmab to the field_table if any domain has progsigma=.true.

**Comment by SamuelTrahanNOAA** [comment]
The C192 forecast completed. Both C384 and C768 are still running.

**Review by lisa-bengtsson — APPROVED** [review]

**Comment by SamuelTrahanNOAA** [comment]
I also want to retest the non-nested configurations before this is merged.

**Comment by SamuelTrahanNOAA** [comment]
The non-nested configurations happily ran to completion.

C96 is being obstinate, so I'll put that in a separate PR.

**Pull request merged**
